### PR TITLE
#166980214 implement pagination

### PR DESCRIPTION
**Pull request by ceexon**
### What does this PR do? 
   - Adds infinite scroll pagination to view all articles.

### Description of Task to be completed?
- Checkout to:
      ft-pagination-166980214

- Install dependencies:

      yarn
     
- Run the local server:

      yarn develop
     
### How should this be manually tested?
- Open the local server URL on your browser and you will see a display of articles
- Scroll down and more articles will continue to unveil

### What are the relevant pivotal tracker stories?
[Pagination](https://www.pivotaltracker.com/story/show/166980214)
### Screenshots
Screenshot 2019-07-24 at 16 43 14
Screenshot 2019-07-24 at 16 43 32
Screenshot 2019-07-24 at 16 43 43

### CODE REVIEW

File: src/components/articles/articlesScroll.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  nextPage,
+  handleClick,
+}) => (
+  <InfiniteScroll
+    data-test="infinite-article-scroll"
+    dataLength={articles.length}
+    next={() => fetchMoreData(nextPage.split('?')[1])}
+    hasMore={!!nextPage}
+    loader={(
+      <div style={{ marginLeft: '90px' }}>
```
Comment (FlevianK) [inline src/components/articles/articlesScroll.js]: In line scss....

File: src/containers/articles/Article.js [hunk 2]
```diff
@@ -5,8 +5,14 @@ import { getArticle, deleteArticle, toggleLike } from '../../actions/articleActi
 import Article from '../../components/articles/Article';
 
 class ArticlePage extends Component {
+  componentWillMount() {
```
Comment (FlevianK) [inline src/containers/articles/Article.js]: What is this working? Because ID in the props is not being update.

File: src/containers/articles/Article.js [hunk 3]
```diff
@@ -5,8 +5,14 @@ import { getArticle, deleteArticle, toggleLike } from '../../actions/articleActi
...
+    const { articleData } = this.props; // eslint-disable-line
+    articleData.id = 0;
+  }
+
   componentDidMount() {
     const { articleData, getArticle, history } = this.props; // eslint-disable-line
+    articleData.id = 0;
```
Comment (FlevianK) [inline src/containers/articles/Article.js:11]: This logic does not make sense to me. 
You have already hard coded the value, it will always be 0 in this function.

File: src/containers/articles/Article.js [hunk 4]
```diff
@@ -5,8 +5,14 @@ import { getArticle, deleteArticle, toggleLike } from '../../actions/articleActi
...
     if (!articleData.id) {
```
Comment (FlevianK) [inline src/containers/articles/Article.js:12]: At what point is this condition `!articleData.id` going to be false?

File: src/reducers/articleReducers.js [hunk 5]
```diff
@@ -37,44 +37,50 @@ export default function (state = initialState, action) {
...
       return {
         ...state,
         actionCalled: true,
-        articles: action.payload.results,
+        articles: state.articles.concat(action.payload.results),
         nextPage: action.payload.next,
         previousPage: action.payload.previous,
         articleCount: action.payload.count,
       };
+
```
Comment (FlevianK) [inline src/reducers/articleReducers.js]: Why the black lines?

File: src/containers/profile/Profile.js [hunk 6]
```diff
@@ -16,6 +16,14 @@ export class UnconnectedProfile extends Component {
     getUserArticles(userId);
   }
 
+  handleArticleClick = (e, slug) => {
+    const { history, getArticle } = this.props; // eslint-disable-line
+    e.preventDefault();
+    const urlTo = `/articles/${slug}`;
+    getArticle(slug);
+    history.push(urlTo);
```
Comment (FlevianK) [inline src/containers/profile/Profile.js:24]: This `getArticle(slug)` should be successful for this  `history.push(urlTo)` to happen. Your logic does not cater for that.

**Review by FlevianK — CHANGES_REQUESTED** [review]
Kindly implement the reviews

File: src/containers/articles/Article.js (hunk 3 above)
Comment (ceexon) [reply src/containers/articles/Article.js:11]: When you click on an article from say from the homepage as it is right now it will open the article and update the global state with the article data. When you open your profile and click on an article you wrote maybe just to view it, the former article is still in the state and shows for some time before the expected article shows up thus I have to set the id to 0 to set the loader  to visible and prevent this instance where a formerly loaded article shows up before the expected one is completely fetched

File: src/containers/articles/Article.js (hunk 4 above)
Comment (ceexon) [reply src/containers/articles/Article.js:12]: As soon as the page renders, the id false is what triggers the loader.

File: src/containers/profile/Profile.js (hunk 6 above)
Comment (ceexon) [reply src/containers/profile/Profile.js:24]: We made this fix in @jonathanmusila 's PR 
I don't think I have to redo it here

File: src/reducers/articleReducers.js (hunk 5 above)
Comment (ceexon) [reply src/reducers/articleReducers.js]: I have removed them
I thought with the blank lines the file looks more readable

File: src/components/__tests__/articles/imageUploader.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  handleChange: mockFn,
+  image: 'my_image_url',
+  imageUrl: '',
+  imageUpload: mockFn,
+};
+
+const getRootProps = () => props;
+const getInputProps = mockFn;
+
+describe('<IconText /> component', () => {
```
Comment (wekesa931) [inline src/components/__tests__/articles/imageUploader.test.js:18]: Good work with the testing!

File: src/components/navbar/Navbar.js [hunk 8]
```diff
@@ -11,9 +11,9 @@ const { Item } = Menu;
 const Navbar = ({ clickedLogin, clickedSignup, logOut }) => (
   <Layout className="nav">
     <Header className="nav">
-      <Link to="/" className="logo">
+      <a href="/" className="logo">
```
Comment (wekesa931) [inline src/components/navbar/Navbar.js:41]: Why did you opt to remove react `Link` property?

**Review by wekesa931 — COMMENTED** [review]
Good work. You inspire me

File: src/components/articles/Featuredarticles.scss [hunk 9]
```diff
@@ -28,7 +28,7 @@ $primary-black: #000;
 }
 
 .featured-title {
-  padding: 12px 0 13%;
+  padding: 12px 0 0 10%;
```
[the remark this answers is not in the document]
Comment (wangonya) [reply src/components/articles/Featuredarticles.scss:31]: You can shorten this to `padding: 12px 0 10%;`

File: src/containers/articles/Article.js (hunk 3 above)
Comment (FlevianK) [reply src/containers/articles/Article.js:11]: cool

**Review by FlevianK — APPROVED** [review]

**Review by wangonya — COMMENTED** [review]
Infinite scroll 🕺🏽🕺🏽

**Review by FlevianK — APPROVED** [review]
